tensorflow/contrib/go/tensor_test.go

```diff
@@ -0,0 +1,630 @@
+					}
+				}
+			}
+		}
+	`)
+
+	result, err := tensor.AsFloat32()
+	if err != nil {
+		t.Error("Problem trying to cast a tensor into float32 slice, Error:", err)
+		t.FailNow()
+	}
+
+	if len(result) != 12 {
+		t.Error("The expected length for the returned slice is 12 but the returned slice length was:", len(result))
+		t.FailNow()
+	}
+
+	for x := 0; x < len(expectedResult); x++ {
+		for y := 0; y < len(expectedResult[x]); y++ {
+			for z := 0; z < len(expectedResult[x][y]); z++ {
+				value, err := tensor.GetVal(x, y, z)
+				if err != nil {
+					t.Error("Error returned when accessing to position:", x, y, z, "Error:", err)
+				}
+				valueFloat := value.(float32)
+				if valueFloat != expectedResult[x][y][z] {
+					t.Errorf(
+						"The expected value for position: %d %d %d is: %f but the one returned was: %f",
+						x,
+						y,
+						z,
+						expectedResult[x][y][z],
+						valueFloat)
+					t.FailNow()
+				}
+			}
+		}
+	}
+}
+
+func TestUint8Decode(t *testing.T) {
+	expectedResult := int32(21)
+	tensor := getTensorFromGraph(t, fmt.Sprintf(`
+		node {
+			name: "output"
+			op: "Const"
+			attr {
+				key: "dtype"
+				value {
+					type: DT_UINT8
+				}
+			}
+			attr {
+				key: "value"
+				value {
+					tensor {
+						dtype: DT_UINT8
+						tensor_shape {
+						}
+						int_val: %d
+					}
+				}
+			}
+		}
+		version: 5`, expectedResult),
+	)
+
+	result, err := tensor.AsInt32()
+	if err != nil {
+		t.Error("Problem trying to cast a tensor into uint8 slice, Error:", err)
+		t.FailNow()
+	}
+
+	if len(result) != 1 {
+		t.Error("The expected length for the returned slice is 1 but the returned slice length was:", len(result))
+		t.FailNow()
+	}
+
+	if result[0] != expectedResult {
+		t.Errorf("The expected value is: %d, but the returned is: %d", expectedResult, result[0])
+		t.FailNow()
+	}
+}
+
+/*func TestUint16Decode(t *testing.T) {
+	expectedResult := uint16(321)
+	tensor := getTensorFromGraph(t, fmt.Sprintf(`
+		node {
+			name: "output"
+			op: "Const"
+			attr {
+				key: "dtype"
+				value {
+					type: DT_UINT16
+				}
+			}
+			attr {
+				key: "value"
+				value {
+					tensor {
+						dtype: DT_UINT16
+						tensor_shape {
+						}
+						int_val: %d
+					}
+				}
+			}
+		}
+		version: 5`, expectedResult),
+	)
+
+	result, err := tensor.AsUint16()
+	if err != nil {
+		t.Error("Problem trying to cast a tensor into uint16 slice, Error:", err)
+		t.FailNow()
+	}
+
+	if len(result) != 1 {
+		t.Error("The expected length for the returned slice is 1 but the returned slice length was:", len(result))
+		t.FailNow()
+	}
+
+	if result[0] != expectedResult {
+		t.Errorf("The expected value is: %d, but the returned is: %d", expectedResult, result[0])
+		t.FailNow()
+	}
+}*/
+
+func TestInt16(t *testing.T) {
+	expectedResult := int32(21)
+	tensor := getTensorFromGraph(t, fmt.Sprintf(`
```

Comment:
There's a lot of repetition of this basic tensor structure that only varies in the type: DT_INTX field.  It might be much more concise to stash the basic template and then sprintf the dtype also if possible.